### PR TITLE
add center to Anchors

### PR DESCRIPTION
**Pull request by oozoofrog**
```swift
self {
  center.anchors {
    Anchors.center()
  }
}
```

### CODE REVIEW

File: Sources/SwiftLayout/Extension/Anchors/Anchors+Convenience.swift [hunk 1]
```diff
@@ -176,4 +176,27 @@ extension Anchors {
         let height = Anchors(.height).equalTo(constant: size.height)
         return width.union(height)
     }
+    
+    /// ``Anchors`` for center to superview
+    /// - Parameters:
+    ///   - offsetX: ``CGFloat`` type. == superview.center.x += offsetX
+    ///   - offsetY: ``CGFloat`` type. == superview.center.y += offsetY
+    /// - Returns: ``Anchors``
+    public static func center<I: ConstraintableItem>(_ toItem:I, offsetX: CGFloat = .zero, offsetY: CGFloat = .zero) -> Anchors {
```
Comment (gmlwhdtjd) [inline Sources/SwiftLayout/Extension/Anchors/Anchors+Convenience.swift:185]: Is this the correct signature?
I think it's different from the description

File: Sources/SwiftLayout/Extension/Anchors/Anchors+Convenience.swift [hunk 2]
```diff
@@ -176,4 +176,27 @@ extension Anchors {
...
+    
+    /// ``Anchors`` for center to item
+    /// - Parameters:
+    ///   - toItem: ``ConstraintableItem`` type,  second item for constraint
+    ///   - offsetX: ``CGFloat`` type. == superview.center.x += offsetX
+    ///   - offsetY: ``CGFloat`` type. == superview.center.y += offsetY
+    /// - Returns: ``Anchors``
+    public static func center<I: ConstraintableItem>(_ toItem:I, offsetX: CGFloat = .zero, offsetY: CGFloat = .zero) -> Anchors {
+        let x = Anchors(.centerX).equalTo(constant: offsetX)
+        let y = Anchors(.centerY).equalTo(constant: offsetY)
```
Comment (gmlwhdtjd) [inline Sources/SwiftLayout/Extension/Anchors/Anchors+Convenience.swift:199]: Is this the correct implementation?
It seems different from the description.

File: Sources/SwiftLayout/Components/Anchors/ConstraintableItem.swift [hunk 3]
```diff
@@ -27,3 +27,4 @@ public protocol ConstraintableItem {}
 extension UIView: ConstraintableItem {}
 extension UILayoutGuide: ConstraintableItem {}
 extension String: ConstraintableItem {}
+extension Never: ConstraintableItem {}
```
Comment (gmlwhdtjd) [inline Sources/SwiftLayout/Components/Anchors/ConstraintableItem.swift:30]: What is the purpose of this?

Comment (oozoofrog) [reply Sources/SwiftLayout/Components/Anchors/ConstraintableItem.swift:30]: no it just mistake 😅

File: Sources/SwiftLayout/Extension/Anchors/Anchors+Convenience.swift [hunk 4]
```diff
@@ -176,4 +176,27 @@ extension Anchors {
         let height = Anchors(.height).equalTo(constant: size.height)
         return width.union(height)
     }
+    
+    /// ``Anchors`` for center to superview
```
Comment (oozoofrog) [inline Sources/SwiftLayout/Extension/Anchors/Anchors+Convenience.swift:180]:
```suggestion
    /// ``Anchors`` for center to item
```

File: Sources/SwiftLayout/Extension/Anchors/Anchors+Convenience.swift (hunk 1 above)
Comment (oozoofrog) [reply Sources/SwiftLayout/Extension/Anchors/Anchors+Convenience.swift:185]: you passed testing, congrat.

File: Sources/SwiftLayout/Extension/Anchors/Anchors+Convenience.swift (hunk 2 above)
Comment (oozoofrog) [reply Sources/SwiftLayout/Extension/Anchors/Anchors+Convenience.swift:199]: human error human error

File: Sources/SwiftLayout/Extension/Anchors/Anchors+Convenience.swift (hunk 1 above)
Comment (oozoofrog) [inline Sources/SwiftLayout/Extension/Anchors/Anchors+Convenience.swift:185]:
```suggestion
    public static func center(offsetX: CGFloat = .zero, offsetY: CGFloat = .zero) -> Anchors {
```

File: Sources/SwiftLayout/Components/Anchors/ConstraintableItem.swift (hunk 3 above)
Comment (oozoofrog) [inline Sources/SwiftLayout/Components/Anchors/ConstraintableItem.swift:30]:
```suggestion
```

File: Sources/SwiftLayout/Extension/Anchors/Anchors+Convenience.swift [hunk 5]
```diff
@@ -176,4 +176,27 @@ extension Anchors {
...
+        let x = Anchors(.centerX).equalTo(constant: offsetX)
+        let y = Anchors(.centerY).equalTo(constant: offsetY)
+        return x.union(y)
+    }
+    
+    /// ``Anchors`` for center to item
+    /// - Parameters:
+    ///   - toItem: ``ConstraintableItem`` type,  second item for constraint
+    ///   - offsetX: ``CGFloat`` type. == superview.center.x += offsetX
+    ///   - offsetY: ``CGFloat`` type. == superview.center.y += offsetY
```
Comment (gmlwhdtjd) [inline Sources/SwiftLayout/Extension/Anchors/Anchors+Convenience.swift:195]:
```suggestion
    ///   - offsetX: ``CGFloat`` type. == item.center.x += offsetX
    ///   - offsetY: ``CGFloat`` type. == item.center.y += offsetY
```

File: Sources/SwiftLayout/Extension/Anchors/Anchors+Convenience.swift [hunk 6]
```diff
@@ -176,4 +176,27 @@ extension Anchors {
...
+    /// - Returns: ``Anchors``
+    public static func center<I: ConstraintableItem>(_ toItem:I, offsetX: CGFloat = .zero, offsetY: CGFloat = .zero) -> Anchors {
+        let x = Anchors(.centerX).equalTo(constant: offsetX)
+        let y = Anchors(.centerY).equalTo(constant: offsetY)
```
Comment (gmlwhdtjd) [inline Sources/SwiftLayout/Extension/Anchors/Anchors+Convenience.swift:199]:
```suggestion
        let x = Anchors(.centerX).equalTo(toItem, constant: offsetX)
        let y = Anchors(.centerY).equalTo(toItem, constant: offsetY)
```

**Review by gmlwhdtjd — APPROVED** [review]
👍

**Pull request merged**